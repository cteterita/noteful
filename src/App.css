header {
  background-color: #999999;
  color: white;
  text-align: center;
}

main {
  display: flex;
}

.error-message {
  color: red;
}
