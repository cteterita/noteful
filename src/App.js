import React from 'react';
import { Route, Link } from 'react-router-dom';
import Context from './Context';
import Sidebar from './Sidebar/Sidebar';
import NoteSidebar from './NoteSidebar/NoteSidebar';
import NoteList from './NoteList/NoteList';
import NoteDetail from './NoteDetail/NoteDetail';
import store from './dummy-store.js';
import './App.css';

class App extends React.Component {
  state = store;
  deleteNote(noteId) {
    //To implement
    console.log(`deleting note ${noteId}`);
  }
  listView(routeProps) {
    const { notes, folders } = this.state;
    const folderId = routeProps.match.params.folderId;
    return (
      <>
        <Sidebar folders={folders} selectedFolderId={folderId} />
        <NoteList notes={folderId ? notes.filter(n => n.folderId === folderId) : notes} />
      </>
    );
  }
  noteView(routeProps) {
    const { notes, folders } = this.state;
    const noteId = routeProps.match.params.noteId;
    const note = notes.find(n => n.id === noteId)
    return (
      <>
        <NoteSidebar
          folder={folders.find(f => f.id === note.folderId )}
          back={routeProps.history.goBack} />
        <NoteDetail {...note} />
      </>
    );
  }
  render() {
    const contextValue = {
      ...store,
      deleteNote: this.deleteNote
    }
    return (
      <>
        <header>
          <h1><Link to="/">Noteful</Link></h1>
        </header>
        <main className="App">
          <Context.Provider value={contextValue}>
            <Route exact path ="/" render={(props) => this.listView(props)} />
            <Route exact path ="/folder/:folderId" render={(props) => this.listView(props)} />
            <Route exact path ="/note/:noteId" render={(props) => this.noteView(props)} />
          </Context.Provider>
        </main>
      </>
    );
  }
}

export default App;
