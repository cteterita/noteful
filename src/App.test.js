import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import FakeData from './dummy-store';

describe('App component', () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><App /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it.skip('renders as expected', () => {
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve(FakeData.folders),
    })); // TODO: Figure out how to mock different fetch instances
    const tree = renderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

jest.restoreAllMocks();
