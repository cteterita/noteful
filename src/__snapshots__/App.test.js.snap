// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders as expected 1`] = `
Array [
  <header>
    <h1>
      <a
        href="/"
        onClick={[Function]}
      >
        Noteful
      </a>
    </h1>
  </header>,
  <span
    className="error-message"
  />,
  <main>
    <section
      className="sidebar"
    >
      <h4>
        No folders found
      </h4>
      <a
        href="/add-folder"
        onClick={[Function]}
      >
        <button
          className="add-folder-button"
          type="submit"
        >
          Add Folder
        </button>
      </a>
    </section>
    <section
      className="NoteList"
    >
      <h4>
        No notes found
      </h4>
      <a
        href="/add-note"
        onClick={[Function]}
      >
        <button
          className="add-note-list"
          type="submit"
        >
          Add Note
        </button>
      </a>
    </section>
  </main>,
]
`;
